Wikipedia/Code/WMFBaseExploreSectionController.m

```diff
@@ -204,61 +216,39 @@ - (AnyPromise*)fetchDataIgnoreError:(BOOL)ignoreError ignoreCurrentItems:(BOOL)i
             }
             return error;
         });
+        // NOTE: set items to placeholders only after promise has been set to indicate isFetching state
+        [self setItemsToPlaceholdersOrEmpty];
         return self.fetcherPromise;
     }
 }
 
 - (void)resetData {
-    self.fetchError = nil;
-    [self setItemsToPlaceholders];
+    [self setItemsToPlaceholdersOrEmpty];
 }
 
 #pragma mark - Utility
 
-- (void)setItemsToPlaceholders {
+- (void)setItemsToPlaceholdersOrEmpty {
     if (![self shouldShowPlaceholderCell]) {
+        self.items = @[];
```

Comment:
@bgerstle can you somehow indicate why you are doing this. Its not clear why we are setting this to an empty array.